test/exec_test.exs

```diff
@@ -382,5 +382,10 @@ defmodule QueryParser.ExecTest do
                %{"_id" => 10}
              ] = parse_and_exec(data(), %{"_id" => %{"$not" => %{"$lt" => 5}}})
     end
+
+    test "size operator can only take array" do
+      assert_raise ArgumentError, fn -> parse_and_exec(data(), %{"_id" => %{"$size" => 5}}) end
```

Comment:
I've tested the $size operator on other fields than array and the data is just filtered just like if it returs 0.